### DISCUSSION TITLE
Looking for option to build a completely vanilla vagrant image

### DISCUSSION THREAD

**Discussion by JoelMiller74** [gusztavvargadr/packer#289] [Q&A] [answered]
I need Vagrant box images for testing in our DevOps pipelines. I love how you have built these images and keep them up to date every month using automation.
However, I run into issues with the fact that the base boxes with just Windows Server also includes chocolatey, OpenSSL, and Chef Client 17. I am testing with Chef Test-Kitchen and currently we require a lower version than newest (17). I wish we were on the newest version, but currently, we are stuck at version 14. When I use test-kitchen with the Windows Server images, I get issues because Chef Client 17 is already installed. I must uninstall it before running my recipes.
I am just looking at finding a way to build using this repo without any additional software like chocolatey, OpenSSL, Chef-Client, etc... on the built box. Test-Kitchen will automatically install the chef client and any additional software I need.
Is there a way to build Windows Server images without any of the Configuration Management tools installed? I am just looking for a vanilla installation.

**Comment by gusztavvargadr** [comment]
@JoelMiller74 thanks for reaching out and sorry for the late reply. Glad to hear you find these images useful, that's why I build them :)

Indeed, I got those tools installed by default as I use all of them for later provisioning. However, it perfectly makes sense to have other scenarios as you describe.

To be able to have plain images as you describe I'd need to keep at least WinRM to control provisioning instead of SSH, but that's required for Test Kitchen also anyway. Would you prefer to have the latest patches installed as well? Then I can think of setting up and publishing plain images at the next release.

Which OS versions and variants would you need?

**Reply by JoelMiller74** [reply]
Thank you very much. Yes. I figured that WinRM configuration would be required regardless. Personally, I would want something completely patched. That is what I like the most about your images. Other images are years old, but yours are always up to date.

As far as what OS and variants, I was thinking that it might be easier to introduce a switch parameter or a config file setting that would allow users to skip things if their build process required items to be skipped. I have seen others use -nocm (no configuration management) to skip Chef, OpenSSL, Chocolatey, etc....

I would  think there might be an easy way to skip all of those things or even allowing the skipping of individual items if needed. Then you would not need to hard-code anything into a specific image, but allow the skipping universally.

**Reply by gusztavvargadr** [reply]
Ok, sure, I was also thinking along the "-nocm" option.

Are you building your own base boxes and / or using the already published ones from https://app.vagrantup.com/gusztavvargadr? Just asking because I have quite a few images and building them takes some time and money already, creating a "-nocm" version for everything is probably not something I can easily afford now. So if you have any preferred images to be published, please let me know, and I'd start with those Of course, if you're building your own, I can make a config option globally available in the build scripts.

For more complex scenarios (like Docker, IIS, SQL Server, Visual Studio) I already have everything provisioned with Chef, I'd rather not go back to pure scripting :) For "-nocm" it can be an option though to provision like now, and uninstall any not needed tools.

So please let me know what you prefer regarding building and publishing.

**Reply by JoelMiller74** [reply]
Currently, we are using the boxes you create on Vagrantup.com. However, we are running into that issue with newer versions of the tools being installed. We would like to build our own images locally and I did not want to reinvent the wheel. Your images seem to use the best process and are built 'always up to date'. I would like to just use your code to build and publish locally to Artifactory. That way, when we pull images, they are from a local repository. 

If you make this change, we would not need you publish all new, additional images to Vagrant. This would just allow us to build it ourselves and publish locally and pull locally.

**Reply by gusztavvargadr** [reply, marked as answer]
Ok, created #296 to track this further. I cannot promis a timeline, but it will be on top of my list :)

Thanks again for reaching out and the suggestions.

**Reply by JoelMiller74** [reply]
Thank you! I appreciate the consideration a lot!
